experimental/tls_attestation/src/server.rs

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::certificate::CertificateAuthority;
+use anyhow::{anyhow, Context};
+use futures::future::FutureExt;
+use futures_util::{future::TryFutureExt, stream::Stream};
+use http::Uri;
+use hyper::{
+    service::{make_service_fn, service_fn},
+    Body, Client, Request, Response, Server,
+};
+use log::warn;
+use std::pin::Pin;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_rustls::{server::TlsStream, TlsAcceptor};
+
+/// Proxies HTTPS requests to an HTTP server specified by `uri`.
+async fn proxy(
+    request: Request<Body>,
+    client: Client<hyper::client::HttpConnector>,
+    uri: http::Uri,
+) -> Result<Response<Body>, hyper::Error> {
+    let (mut parts, body) = request.into_parts();
+    parts.uri = uri;
+
+    let proxy_request = Request::from_parts(parts, body);
+    client.request(proxy_request).await
+}
+
+/// Run HTTPS proxy server that provides self-signed TLS certificates with TEE attestation
```

Comment:
```suggestion
/// Runs HTTPS proxy server that provides self-signed TLS certificates with TEE attestation
```
Perhaps also briefly describe what `domain_name` is.